Packs/Druva/Integrations/DruvaEventCollector/DruvaEventCollector.py

```diff
@@ -0,0 +1,289 @@
+            if "invalid_grant" in str(e):
+                raise DemistoException(
+                    "Error in test-module: Make sure Server URL, Client ID and Secret Key are correctly entered."
+                ) from e
+            raise
+        return response_json["access_token"], response_json["expires_in"]
+
+    def search_events(self, tracker: Optional[str] = None) -> dict:
+        """
+        Searches for Druva events.
+
+        Args:
+            tracker: pointer to the last event we got last time
+
+        Returns:
+            dict: List of events
+        """
+        demisto.debug(f'DEC: This is the tracker before encoding: {tracker=}')
+
+        if tracker:
+            encoding_tracker = quote(tracker, safe="!~*'()")  # remove invalid characters from tracker
+            demisto.debug(f'DEC: This is the tracker after encoding: {encoding_tracker=}')
+        else:
+            encoding_tracker = ""
+
+        url_suffix_tracker = f"?tracker={encoding_tracker}" if encoding_tracker else ""
+
+        headers = (self._headers or {}) | {
+            "accept": "application/json"
+        }  # self._headers won't really be None, just for mypy
+        try:
+            response = self._http_request(
+                method="GET",
+                url_suffix=f"/insync/eventmanagement/v2/events{url_suffix_tracker}",
+                headers=headers,
+            )
+        except Exception as e:
+            # 403 - "User is not authorized to access this resource with an explicit deny" - reason: tracker is expired
+            # 400 - "Invalid tracker"
+            raise DemistoException(f"Error in search-events: {str(e)}") from e
+        return response
+
+    def _set_headers(self, token: str):
+        self._headers = {"Authorization": f"Bearer {token}"}
+
+    def _max_fetch_validation(self):
+        if self.max_fetch > MAX_FETCH or self.max_fetch < MIN_FETCH:
+            raise DemistoException(f"The maximum number of events per fetch should be between 1 - {MAX_FETCH}")
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests API connectivity and authentication
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+
+    Args:
+        client (Client): Druva client to use.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    get_events(client=client)
+    return "ok"
+
+
+def get_events(client: Client, tracker: Optional[str] = None) -> tuple[list[dict], str]:
+    """
+    Gets events from Druva API in one batch (max 500), if a tracker is given, the API returns events starting from its timestamp.
+    Args:
+        client: Druva client to use.
+        tracker: A string received in a previous run, marking the point in time from which we want to fetch.
```

Comment:
worth mentioning what's the tracker value when there are no new values to fetch (same one if I recall correctly)